upgrade translation for control-plane-node-communication (#22303)
kubernetes/website: content/it/docs/concepts/architecture/control-plane-node-communication.md

---
title: Comunicazione Piano di controllo - Nodo
content_type: concept
weight: 20
aliases:
- master-node-communication
---

<!-- overview -->

> Questo documento cataloga i percorsi di comunicazione tra il piano di controllo (_control plane_), in realtà l'apiserver, e il cluster Kubernetes. L'intento è di consentire agli utenti di personalizzare la loro installazione per rafforzare la configurazione di rete affinché il cluster possa essere eseguito su una rete non affidabile (o su IP completamente pubblici resi disponibili da un fornitore di servizi cloud).

Reviewer: ```suggestion
Questo documento cataloga le connessioni tra il piano di controllo (_control plane_), in realtà l'apiserver, e il cluster Kubernetes. L'intento è di consentire agli utenti di personalizzare la loro installazione affinché il cluster possa essere eseguito su una rete publica (o su IP resi disponibili da un fornitore di servizi cloud).
```
Proposing some changes to make it more fluent...

Author: fixed

<!-- body -->

> ## Dal Nodo al piano di controllo{#dal-nodo-al-piano-di-controllo}

Reviewer: ```suggestion
## Dal Nodo al control-plane{#dal-nodo-al-piano-di-controllo}
```
Once an english term is defined, IMO we should use it to make the document less awkward for users acquainted to reading Kubernetes docs/articles

Author: I agree and I change it accordingly but I understood from my previous translation that we should translate them all to the local language adding in parenthesis the en/standard word/expression only at the first occurrence

Reviewer: My understanding that once an english term is defined we can use that without repeating the translitions

Reviewer 2: I'm wondering what a Spanish speaker gets from having “control-plane” in the text rather than “piano di controllo”, which seems to me like the right term to use throughout.

If the original were in Spanish I wouldn't expect to see piano-di-controllo repeated through the English version of the document.

Reviewer: @sftim I got several feedbacks from users telling us that exceeding in translating mainstream terms in Italian only creates confusion and makes the documentation look weird. 

It is a fact that most of the documentation is in English, most of the blogs are in English and Italian users working in IT are using a jargon of Italian and English mainstream terms (and this happens not only in Kubernetes).

Se, I think that as a contributor my goal is to meet the need of our users base finding the right balance between literal translation - which makes docs look weird - and the jargon IT users are used to. 

With regards to the hypothetical discussion about what if the original term was "piano di controllo", I would say that in real life this is not the case. At the end also Ensglish native speaker uses italian terms like pizza or pasta without translating them.

Reviewer 2: Based on https://github.com/kubernetes/website/pull/22303#discussion_r453828496, does it make sense to use “control plane” in the page title also and then explain that term as an aside in the page introduction?

Author: ok I'll change the title as well using "control-plane". I'll use "piano di controllo" (control-plane) within the page at the occurrence and then always "control-plane". Deal?

Reviewer 2: (sounds good to me - happy for Italian localization approvers to have a final say)

Author: done

> Kubernetes adotta un pattern per le API di tipo _"hub-and-spoke"_. Tutte le chiamate delle API eseguite sui vari nodi sono effettuate verso l'apiserver (nessuno degli altri componenti principali è progettato per esporre servizi remoti). L'apiserver è configurato per l'ascolto di connessioni remote su una porta HTTPS protetta (443) con una o più forme di [autenticazioni](/docs/reference/access-authn-authz/authentication/) _client_ abilitate. Si dovrebbero abilitare una o più forme di [autorizzazioni](/docs/reference/access-authn-authz/authorization/), in particolare nel caso in cui siano ammesse [richieste anonime](/docs/reference/access-authn-authz/authentication/#anonymous-requests) o [_token_ legati ad un account di servizio (_service account_)](/docs/reference/access-authn-authz/authentication/#service-account-tokens).

Reviewer: ```suggestion
Kubernetes adotta un pattern per le API di tipo _"hub-and-spoke"_. Tutte le chiamate delle API eseguite sui vari nodi sono effettuate verso l'apiserver (nessuno degli altri componenti principali è progettato per esporre servizi remoti). L'apiserver è configurato per l'ascolto di connessioni remote su una porta HTTPS protetta (tipicamente 443) con una o più forme di [autenticazione client](/docs/reference/access-authn-authz/authentication/) abilitate. Si dovrebbero abilitare una o più forme di [autorizzazioni](/docs/reference/access-authn-authz/authorization/), in particolare nel caso in cui siano ammesse [richieste anonime](/docs/reference/access-authn-authz/authentication/#anonymous-requests) o [_token_ legati ad un account di servizio (_service account_)](/docs/reference/access-authn-authz/authentication/#service-account-tokens).
```

Author: fixed

> Il certificato pubblico di radice (_public root certificate_) relativo al cluster corrente deve essere fornito ai vari nodi di modo che questi possano connettersi in modo sicuro all'apiserver insieme alle  credenziali valide per uno specifico _client_. Ad esempio, nella configurazione predefinita di un cluster [GKE](https://cloud.google.com/kubernetes-engine?hl=it), le credenziali del client fornite al kubelet hanno la forma di un certificato client. Si veda

Reviewer: ```suggestion
Il certificato pubblico (_public root certificate_) relativo al cluster corrente deve essere fornito ai vari nodi di modo che questi possano connettersi in modo sicuro all'apiserver insieme alle  credenziali valide per uno specifico _client_. Ad esempio, nella configurazione predefinita di un cluster [GKE](https://cloud.google.com/kubernetes-engine?hl=it), le credenziali del client fornite al kubelet hanno la forma di un certificato client. Si veda
```

Author: fixed

[inizializzazione TLS del kubelet TLS](/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/) per la fornitura automatica dei certificati client al _kubelet_.

> I _pod_ che desiderano connettersi all'apiserver possono farlo in modo sicuro sfruttando un account di servizio in modo che Kubernetes inserisca automaticamente il certificato pubblico di radice e un token valido al portatore (_bearer token_) all'interno _pod_ quando questo viene istanziato.

Reviewer 2: nit (OK to skip fixing this, but nice to have):
```suggestion
I _Pod_ che desiderano connettersi all'apiserver possono farlo in modo sicuro sfruttando un account di servizio in modo che Kubernetes inserisca automaticamente il certificato pubblico di radice e un token valido al portatore (_bearer token_) all'interno _Pod_ quando questo viene istanziato.
```
because pod is not a common word in Italian.

Author: ok to fix it nop. English term will be always Pascal-Case regardless the context or that rule is only related to "k8s Resource"?

Reviewer 2: In _English_ I can choose between (eg) “Service” (the resource name) and “service” and not have to worry too much about which I mean. For a localization you usually have to be a little more precise because the localized term has a different name.

Author: according to this rule, the section "### Dall'apiserver ai nodi, Pod, e servizi{#dal-apiserver-ai-nodi-pod-servizi}" should use "Service" (as k8s resource) instead of its the translation "servizi" and so on within the description. @sftim what do you think?

Reviewer 2: I wasn't sure whether the English original was talking about services or Services, so I don't mind either way.

Reviewer 2: Does the API server ever send messages _to_ a Service?

> La risorsa _Service_ di `kubernetes` (in tutti i _namespace_) è configurata con un indirizzo IP virtuale che viene reindirizzato (tramite _kube-proxy_) all'endpoint HTTPS dell'apiserver.

Reviewer: ```suggestion
In tutti i namespace è configurato un _Service_ con nome `kubernetes` con un indirizzo IP virtuale che viene reindirizzato (tramite _kube-proxy_) all'endpoint HTTPS dell'apiserver.
```

Author: fixed

> Anche i componenti del piano d controllo comunicano con l'apiserver del cluster su di una porta sicura esposta da quest'ultimo.

Reviewer: ```suggestion
Anche i componenti del control-plane comunicano con l'apiserver del cluster su di una porta sicura esposta da quest'ultimo.
```

Author: done

Di conseguenza, la modalità operativa predefinita per le connessioni dai nodi e dai _pod_ in esecuzione sui nodi verso il piano di controllo è protetta da un'impostazione predefinita
e può essere eseguita su reti non sicure e/o pubbliche.

> ## Dal piano di controllo al nodo{#dal-piano-di-controllo-al-nodo}

Reviewer: ```suggestion
## Dal control-plane al nodo{#dal-piano-di-controllo-al-nodo}
```

Author: done

> Esistono due percorsi di comunicazione principali dal piano di controllo (apiserver) verso i nodi. Il primo è dall'apiserver verso il processo _kubelet_ in esecuzione su ogni nodo nel cluster. Il secondo è dall'apiserver a ciascun nodo, _pod_, o servizio attraverso la funzionalità proxy dell'apiserver.

Reviewer: ```suggestion
Esistono due percorsi di comunicazione principali dal control-plane (apiserver) verso i nodi. Il primo è dall'apiserver verso il processo _kubelet_ in esecuzione su ogni nodo nel cluster. Il secondo è dall'apiserver a ciascun nodo, _pod_, o servizio attraverso la funzionalità proxy dell'apiserver.
```

Author: done

### Dall'apiserver al _kubelet_

Le connessioni dall'apiserver al _kubelet_ vengono utilizzate per:

>   * Prendere i log relativi ai vari _pod_.
>   * Collegarsi (attraverso kubectl) ai pod in esecuzione.

Reviewer 2: nit (OK to skip fixing this, but nice to have):
```suggestion
  * Prendere i log relativi ai vari _Pod_.
  * Collegarsi (attraverso kubectl) ai Pod in esecuzione.
```
because pod is not a common word in Italian and, here, clearly refers to the API object.

Author: ok I will change it according the rule we agreed above for k8s resources

Author: just a not... if the word refers to k8s resource will have first letter uppercase but do I have to skip the italic in this case?

  * Fornire la funzionalità di _port-forwarding_ per i _kubelet_.

> Queste connessioni terminano all'endpoint HTTPS del _kubelet_. Di default, l'apiserver non verifica il certificato servito dal _kubelet_, il che rende la connessione soggetta ad attacchi _man-in-the-middle_, e tale  da essere considerato **non sicuro (unsafe)** se eseguito su reti non protette e/o pubbliche.

Reviewer 2: If a reader sees “**non sicuro**” without the “(unsafe)” text, is the meaning clear enough?

> Queste connessioni terminano all'endpoint HTTPS del _kubelet_. Di default, l'apiserver non verifica il certificato servito dal _kubelet_, il che rende la connessione soggetta ad attacchi _man-in-the-middle_, e tale  da essere considerato **non sicuro (unsafe)** se eseguito su reti non fidate e/o pubbliche.

Reviewer: ```suggestion
Queste connessioni terminano all'endpoint HTTPS del _kubelet_. Di default, l'apiserver non verifica il certificato servito dal _kubelet_, il che rende la connessione soggetta ad attacchi _man-in-the-middle_, e tale  da essere considerato **non sicuro (unsafe)** se eseguito su reti non protette e/o pubbliche.
```

Author: fixed

Per verificare questa connessione, si utilizzi il parametro `--kubelet-certificate-authority` al fine di fornire all'apiserver un insieme di certificati radice da utilizzare per verificare il
il certificato servito dal _kubelet_.

> Se questo non è possibile, si usi un [tunnel SSH](/docs/tasks/access-application-cluster/port-forward-access-application-cluster/) tra l'apiserver e il _kubelet_, se richiesto, per evitare il collegamento su una rete non fidata o pubblica.

Reviewer: ```suggestion
Se questo non è possibile, si usi un [tunnel SSH](/docs/tasks/access-application-cluster/port-forward-access-application-cluster/) tra l'apiserver e il _kubelet_, se richiesto, per evitare il collegamento su una rete non protetta o pubblica.
```

Author: fixed

In fine, l'[autenticazione e/o l'autorizzazione del kubelet](/docs/admin/kubelet-authentication-authorization/) dovrebbe essere abilitate per proteggere le API esposte dal _kubelet_.

### Dall'apiserver ai nodi, _pod_, e _servizi_{#dal-apiserver-ai-nodi-pod-servizi}

> Le connessioni dall'apiserver verso un nodo, _pod_ o servizio avvengono in modalità predefinita su semplice connessione HTTP e quindi non sono né autenticate né criptata. Queste connessioni possono essere eseguite su una connessione HTTPS sicura mediante il prefisso `https:` al nodo, pod o nome del servizio nell'URL dell'API, ma non valideranno il certificato fornito dall'endpoint HTTPS né forniranno le credenziali del client così anche se la connessione verrà criptata, non fornirà alcuna garanzia di integrità. **Non è attualmente sicuro** eseguire queste connessioni  su reti non fidate e/o pubbliche.

Reviewer: ```suggestion
Le connessioni dall'apiserver verso un nodo, _pod_ o servizio avvengono in modalità predefinita su semplice connessione HTTP e quindi non sono né autenticate né criptata. Queste connessioni possono essere eseguite su una connessione HTTPS sicura mediante il prefisso `https:` al nodo, pod o nome del servizio nell'URL dell'API, ma non valideranno il certificato fornito dall'endpoint HTTPS né forniranno le credenziali del client così anche se la connessione verrà criptata, non fornirà alcuna garanzia di integrità. **Non è attualmente sicuro** eseguire queste connessioni su reti non protette e/o pubbliche.
```

Author: fixed

### I tunnel SSH

> Kubernetes supporta i _tunnel_ SSH per proteggere la comunicazione tra il piano di controllo e i nodi. In questa configurazione, l'apiserver inizializza un tunnel SSH con ciascun nodo del cluster (collegandosi al server SSH in ascolto sulla porta 22) e fa passare tutto il traffico verso il _kubelet_, il nodo, il _pod_, o il servizio attraverso questo tunnel. Questo tunnel assicura che il traffico non sia esposto al di fuori della rete su cui sono in esecuzioni i vari nodi.

Reviewer: ```suggestion
Kubernetes supporta i _tunnel_ SSH per proteggere la comunicazione tra il control-plane e i nodi. In questa configurazione, l'apiserver inizializza un tunnel SSH con ciascun nodo del cluster (collegandosi al server SSH in ascolto sulla porta 22) e fa passare tutto il traffico verso il _kubelet_, il nodo, il _pod_, o il servizio attraverso questo tunnel. Questo tunnel assicura che il traffico non sia esposto al di fuori della rete su cui sono in esecuzioni i vari nodi.
```

Author: done

I tunnel SSH sono al momento deprecati ovvero non dovrebbero essere utilizzati a meno che ci siano delle esigenze particolari. Il servizio `Konnectivity` è pensato per rimpiazzare questo canale di comunicazione.

### Il servizio _Konnectivity_

{{< feature-state for_k8s_version="v1.18" state="beta" >}}

Come rimpiazzo dei tunnel SSH, il servizio _Konnectivity_ fornisce un proxy a livello TCP per la comunicazione tra il _control-plane_ e il cluster. Il servizio _Konnectivity_ consiste in due parti: il _Konnectivity_ server e gli agenti _Konnectivity_, in esecuzione rispettivamente sul _control-plane_ e sui vari nodi.  Gli agenti _Konnectivity_ inizializzano le connessioni verso il server _Konnectivity_ e mantengono le connessioni di rete. Una volta abilitato il servizio _Konnectivity_, tutto il traffico tra il _control-plane_ e i nodi passa attraverso queste connessioni.

> Si può fare riferimento al [tutorial per il servizio _Konnectivity_](/docs/tasks/extend-kubernetes/setup-konnectivity/) per configurare il servizio _Konnectivity_ all'interno del cluster

Reviewer 2: Would an Italian term such as “lezione” be more suitable than “tutorial”?

Author: ok to translate also "tutorial" but IMHO in Italian tutorial != lezione. Eventually I could use "guida passo-passo"

Reviewer 2: Fair enough, the loanword “tutorial” seems current.

> Si usi il [tutorial per il servizio `Konnectivity`](/docs/tasks/extend-kubernetes/setup-konnectivity/) per configurare il servizio `Konnectivity` all'interno del cluster

Reviewer 2: I recommend against writing Konnectivity in backticks - that makes it look like a command line you might invoke.

Author: backticks removed

> Si usi il [tutorial per il servizio `Konnectivity`](/docs/tasks/extend-kubernetes/setup-konnectivity/) per configurare il servizio `Konnectivity` all'interno del cluster

Reviewer: ```suggestion
Si può fare riferimento [tutorial per il servizio `Konnectivity`](/docs/tasks/extend-kubernetes/setup-konnectivity/) per configurare il servizio `Konnectivity` all'interno del cluster
```

Author: done
